internal/internal_workflow.go

```diff
@@ -567,17 +580,35 @@ func (c *channelImpl) receiveAsyncImpl(callback receiveCallback) (v interface{},
 	return nil, false, true
 }
 
+func (c *channelImpl) removeReceiveCallback(callback *receiveCallback) {
```

Comment:
See the last update at https://vbauerster.github.io/2017/04/removing-items-from-a-slice-while-iterating-in-go/ for removing without allocating.

Reply:
we only remove one element, so the current way should already be the optimized one as it reuses the same underlying array.